packages/frontend/src/features/IoTDeviceCatalogue/ProductTable.tsx

```diff
@@ -0,0 +1,88 @@
+import {
+  Box,
+  TableContainer,
+  Table,
+  Thead,
+  Tr,
+  Th,
+  Tbody,
+  Checkbox,
+} from "@chakra-ui/react";
+import TableRow from "./TableRow";
+import { ProductsSchema } from "../../api/generated";
+
+interface IProductTableProps {
+  isLightMode: boolean;
+  getProducts: {
+    data: Array<ProductsSchema>;
+  };
+  selectedItems: string[];
+  setSelectedItems: React.Dispatch<React.SetStateAction<string[]>>;
+}
+
+const ProductTable: React.FC<IProductTableProps> = ({
```

Comment:
have a look at `@tanstack/table`. We can make a generic datatable to use across all components:
See example [here](https://github.com/Team3132/AttendanceSystem/blob/master/packages/frontend/src/components/DataTable.tsx)

Reply:
defer